src/buildMiddleware.ts

```diff
@@ -70,11 +71,53 @@ export function buildMiddleware<Definitions extends EndpointDefinitions, Reducer
       startNextPoll(action.meta.arg, mwApi);
     }
 
+    if (onFocus.match(action)) {
+      refetchValidQueries(mwApi, 'refetchOnFocus');
+    }
+    if (onOnline.match(action)) {
+      refetchValidQueries(mwApi, 'refetchOnReconnect');
+    }
+
     return result;
   };
 
   return { middleware };
 
+  function refetchValidQueries(api: MWApi, type: 'refetchOnFocus' | 'refetchOnReconnect') {
+    const state = api.getState()[reducerPath];
+    const queries = state.queries;
+    const subscriptions = state.subscriptions;
+    const baseTypeValue = state.config[type]; // The defaults are set in buildSlice when calling createApi
+
+    batch(() => {
+      for (const queryCacheKey of Object.keys(subscriptions)) {
+        const querySubState = queries[queryCacheKey];
+
+        const allSubsAreFalse = Object.values(subscriptions[queryCacheKey] || {}).every(
+          (entry) => type in entry && !entry[type]
+        );
+
+        const shouldRefetch = (baseTypeValue && !allSubsAreFalse) || !allSubsAreFalse;
```

Comment:
Hmm. `(a && b) || b` -> is that not always `b`?
Meaning the global default would be ignored at this point?

Reply:
Updated, but probably not in a great way. Would appreciate any cleanup help here!